### PR TITLE
received unit can keep no, all, or movement orders

### PR DESCRIPTION
**Pull request by loveridge**
Changes the default for received units to only remove all non movement commands from their command queue. This will let shared units continue moving towards their original order location. Also adds settings to customize behavior to stop all commands, or allow all commands after receiving a unit.

This MR is contingent on https://github.com/beyond-all-reason/spring/pull/996

### CODE REVIEW

File: luaui/Widgets/unit_handle_orders_on_share.lua [hunk 1]
```diff
@@ -0,0 +1,90 @@
+function widget:GetInfo()
+	return {
+		name    = "Handle Orders on Share",
+		desc    = "Handles command queue of shared units",
+		author  = "lov",
+		date    = "September 2023",
+		license = "GNU GPL, v2 or later",
+		layer   = 1,
```
Comment (sprunk) [inline luaui/Widgets/unit_handle_orders_on_share.lua:8]: why 1?

Comment (loveridge) [reply luaui/Widgets/unit_handle_orders_on_share.lua:8]: No reason, could be 0

File: language/en/interface.json [hunk 2]
```diff
@@ -194,7 +194,7 @@
 			"morph_tooltip": "Upgrade to next Tech-level (second click to cancel)",
 			"Spawning Disabled": "Spawning disabled",
 			"Spawning Enabled": "Spawning enabled",
-			
+
```
Comment (salinecitrine) [inline language/en/interface.json:197]:
```suggestion

```

extra line

File: luarules/gadgets/unit_cancel_orders_on_share.lua [hunk 3]
```diff
@@ -1,53 +1,33 @@
...
-        name      = "Cancel orders on share",
-        desc      = "Prevents units carrying on with orders once shared/taken and turns on mexes that have been captured",
-        author    = "Bluestone, Beherith",
-        date      = "Jan 2015",
-        license   = "GNU GPL, v2 or later",
-        layer     = 0,
-        enabled   = true
-    }
+	return {
+		name    = "Cancel orders on share",
```
Comment (salinecitrine) [inline luarules/gadgets/unit_cancel_orders_on_share.lua:3]: since this gadget no longer does "Cancel orders on share", it should be renamed

**Review by salinecitrine — APPROVED** [review]
Tested this and works fine for me

**Comment by thehobojoe** [comment]
@loveridge is this PR still relevant? Shared units keep their commands as of an update a while ago, though now that I'm looking for it I can't find the exact commit that made that change

**Comment by sprunk** [comment]
> can't find the exact commit that made that change

engine

**Comment by loveridge** [comment]
There were a couple of threads about units dropping orders when shared on discord, and there was a concern that it could enable griefing. 

https://discord.com/channels/549281623154229250/1149219652665348096
https://discord.com/channels/549281623154229250/1144238387621015692

However the engine was changed to give control over this behavior to gameside back in September. So I guess it would be a question for moderators if it is still necessary to clear some orders when sharing units

**Comment by salinecitrine** [comment]
> @loveridge is this PR still relevant? Shared units keep their commands as of an update a while ago, though now that I'm looking for it I can't find the exact commit that made that change

https://github.com/beyond-all-reason/Beyond-All-Reason/issues/2328 has a lot of context

**Pull request closed without merging**